cupy/linalg/decomposition.py

```diff
@@ -52,13 +50,32 @@ def cholesky(a):
         cusolver.spotrf(
             handle, cublas.CUBLAS_FILL_MODE_UPPER, n, x.data.ptr, n,
             workspace.data.ptr, buffersize, dev_info.data.ptr)
-    else:  # dtype == 'd'
+    elif dtype == 'd':
         buffersize = cusolver.dpotrf_bufferSize(
             handle, cublas.CUBLAS_FILL_MODE_UPPER, n, x.data.ptr, n)
         workspace = cupy.empty(buffersize, dtype=numpy.float64)
         cusolver.dpotrf(
             handle, cublas.CUBLAS_FILL_MODE_UPPER, n, x.data.ptr, n,
             workspace.data.ptr, buffersize, dev_info.data.ptr)
+    elif dtype == 'F':
+        buffersize = cusolver.cpotrf_bufferSize(
+            handle, cublas.CUBLAS_FILL_MODE_UPPER, n, x.data.ptr, n)
+        workspace = cupy.empty(buffersize, dtype=numpy.complex64)
+        cusolver.cpotrf(
+            handle, cublas.CUBLAS_FILL_MODE_UPPER, n, x.data.ptr, n,
+            workspace.data.ptr, buffersize, dev_info.data.ptr)
+    elif dtype == 'D':
+        buffersize = cusolver.zpotrf_bufferSize(
+            handle, cublas.CUBLAS_FILL_MODE_UPPER, n, x.data.ptr, n)
+        workspace = cupy.empty(buffersize, dtype=numpy.complex128)
+        cusolver.zpotrf(
+            handle, cublas.CUBLAS_FILL_MODE_UPPER, n, x.data.ptr, n,
+            workspace.data.ptr, buffersize, dev_info.data.ptr)
+    else:
+        raise ValueError(
+            'dtype must be int32, int64, uint32, uint64, float32, float64,'
+            ' complex64 or complex128 (actual: {})'.format(dtype))
```

Comment:
I didn't get when the error is raised.  Because of `dtype = numpy.find_common_type(...)` above, `dtype` is in `[float32, float64, complex64, complex128]`.  By the way, `cupy.linalg.cholesky` should fail for `float16` inputs as NumPy does, which could be fixed in a new PR.

Reply:
For the record, talked with @toslunar offline and agreed to simply assert that the dtype here is among the expected ones, i.e. not raise an explicit error here. We should fix the dtype behavior not just in this function but all similar (linalg) functions in a separate issue.